### PR TITLE
CT 116: Update EDU Program File Processing and Data Display #19956

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
As a developer, I need to update the ingestion of the EDU Program file to handle null values in most fields so that the file can be uploaded and the UI is not negatively impacted.

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19956

## Acceptance criteria
- [ ] 1. The EDU Program file can be uploaded to GIDS with ANY of the file fields being null or empty.
- [ ] 2. If any of the following fields are null or empty on upload a warning message is displayed stating that the row will not be able to be matched with a corresponding row:
     - Facility Code
     - VET TEC Program
- [ ] 3. Only data with both of the following rows is included in the preview generation and publish actions:
     - Facility Code
     - VET TEC Program

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Comment by zurbergram** [comment]
Needs https://github.com/department-of-veterans-affairs/gibct-data-service/pull/494 to be merged first

**Review by delli-gatti-michael — APPROVED** [review]
Approved by QA

**Review by annaswims — DISMISSED** [review]
please make migration changes in a separate PR

File: app/validators/edu_program_validator.rb [hunk 1]
```diff
@@ -5,7 +5,8 @@ def validate(record)
     duplicates_exist = EduProgram.where(['vet_tec_program = ? AND facility_code = ? AND id != ?',
                                          record.vet_tec_program, record.facility_code, record.id]).any?
     record.errors[:base] << non_unique_error_msg(record) if duplicates_exist
-
+    record.errors[:base] << 'The Facility Code is blank:' if record.facility_code.blank?
+    record.errors[:base] << 'The VET TEC Program (Program Name) is blank:' if record.vet_tec_program.blank?
```
Comment (jholton) [inline app/validators/edu_program_validator.rb]: Can we get some specs in `edu_program_validator_spec` covering these cases?

File: app/models/institution_builder.rb [hunk 2]
```diff
@@ -535,7 +535,7 @@ def self.build_institution_programs(version_number)
         full_time_undergraduate,
         graduate,
         full_time_modifier,
-        length,
+        length_in_weeks,
```
Comment (jholton) [inline app/models/institution_builder.rb]: Why this change? This is populating `institution_programs.length_in_hours` and we already have `length_in_weeks` populating `institution_programs.length_in_weeks` in this SQL (see line 551).

Comment (zurbergram) [reply app/models/institution_builder.rb]: This is a short term fix to an issue that they aren't sure we should fix

**Comment by zurbergram** [comment]
> please make migration changes in a separate PR

https://github.com/department-of-veterans-affairs/gibct-data-service/pull/497

File: app/validators/edu_program_validator.rb (hunk 1 above)
Comment (zurbergram) [reply app/validators/edu_program_validator.rb]: added

File: app/models/institution_builder.rb (hunk 2 above)
Comment (jholton) [reply app/models/institution_builder.rb]: Can you add a comment or link to an issue in the code so that other engineers aren't confused about it later?

**Review by jholton — APPROVED** [review]
